src/hmem_cuda.c

```diff
@@ -484,29 +492,46 @@ static int cuda_hmem_verify_devices(void)
 	 * call to cudaGetDeviceCount() when possible.
 	 */
 
-	/* Make certain that the NVML routines are initialized */
-	nvml_ret = ofi_nvmlInit_v2();
-	if (nvml_ret != NVML_SUCCESS)
-                return -FI_ENOSYS;
+	/* If we are dlopen'ing the CUDA libraries, allow for the case
+	 * where the CUDA libraries are present, but the NVML library is
+	 * not present.  Set nvml_device_count to 1 in that case, bypassing
+	 * actual device count check.
+	 */
+#if ENABLE_CUDA_DLOPEN
+	if (!cuda_attr.nvml_handle)
+		nvml_device_count = 1;
+#endif
 
-	/* Verify NVIDIA devices are present on the host. */
-	nvml_ret = ofi_nvmlDeviceGetCount_v2(&nvml_device_count);
-	if (nvml_ret != NVML_SUCCESS) {
-	        ofi_nvmlShutdown();
-                return -FI_ENOSYS;
-        }
+	/* Check for NVIDIA devices, if NVML library is dlopen-ed*/
+	if (nvml_device_count == 0) {
```

Comment:
This check should be ```if(cuda_attr.nvml_handle)```, which directly captures what's needed prior to executing the selected code.  Don't use a proxy variable.